client/state/analytics/middleware.js

```diff
@@ -32,13 +32,18 @@ const eventServices = {
 	adwords: ( { properties } ) => trackCustomAdWordsRemarketingEvent( properties ),
 };
 
-// Whitelists specific parameters to avoid polluting page view events
-const PAGE_VIEW_SERVICES_ALLOWED_PARAMS = [ 'client_id' ];
+// list of unsafe params that need to be blocked from beign passed down to the recorder.
+const PAGE_VIEW_SERVICES_BLOCKED_PARAMS = [ 'service' ];
```

Comment:
On that note, are there other reserved values we should not override for analytics reporting? Blacklists are typically more dangerous than whitelists since we need knowledge about everything we should block vs knowing what's safe.

Reply:
I dug a bit further to see where client_id was added in https://github.com/Automattic/wp-calypso/pull/20502 

Overall I think we're okay with this or further destructuring before this method, since a plain tracks call will blindly report values 👍